### PR TITLE
Correctly show descriptions for each scope

### LINKED ISSUE #183231: scope should not be exposed on the API EnvironmentVariableMutator object
We need to be strict with API objects:

https://github.com/microsoft/vscode/blob/36b7d5beb3f77fadd6f77d342a0b8b5366cbd12f/extensions/vscode-api-tests/src/singlefolder-tests/terminal.test.ts#L852-L864

### PR DESCRIPTION
**Pull request by karrtikr**
Follow-up to https://github.com/microsoft/vscode/pull/182238

Closes https://github.com/microsoft/vscode/issues/183231

### CODE REVIEW

File: extensions/vscode-api-tests/src/singlefolder-tests/terminal.test.ts [hunk 1]
```diff
@@ -843,24 +843,40 @@ import { assertNoRpc, poll } from '../utils';
 			});
 
 			test('get and forEach should work', () => {
-				const collection = extensionContext.environmentVariableCollection;
+				const collection = extensionContext.environmentVariableCollection as (EnvironmentVariableCollection & { getScopedEnvironmentVariableCollection(scope: EnvironmentVariableScope): EnvironmentVariableCollection });
```
Comment (Tyriar) [inline extensions/vscode-api-tests/src/singlefolder-tests/terminal.test.ts:846]: Add `// TODO` to remove once API is stabilized

File: src/vs/workbench/contrib/terminal/browser/environmentVariableInfo.ts [hunk 2]
```diff
@@ -109,6 +93,28 @@ export class EnvironmentVariableInfoChangesActive implements IEnvironmentVariabl
 	}
 }
 
+function getMessage(collection: IMergedEnvironmentVariableCollection, scope: EnvironmentVariableScope | undefined, extensionService: IExtensionService, extSet: Set<string>): string {
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/environmentVariableInfo.ts:96]: `getMergedDescription`?

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/vs/workbench/api/common/extHostTerminalService.ts [hunk 3]
```diff
@@ -913,7 +916,7 @@ class EnvironmentVariableCollection {
 
 	private _setIfDiffers(variable: string, mutator: vscode.EnvironmentVariableMutator & { scope: vscode.EnvironmentVariableScope | undefined }): void {
 		if (!mutator.scope) {
-			delete (mutator as any).scope; // Convenient for tests
+			delete mutator.scope; // Convenient for tests
 		}
```
[the remark this answers is not in the document]
Comment (Tyriar) [reply src/vs/workbench/api/common/extHostTerminalService.ts:920]: When working on options just now I wanted to make this change:

```
	private _setIfDiffers(variable: string, mutator: vscode.EnvironmentVariableMutator, scope: vscode.EnvironmentVariableScope | undefined): void {
```

Is it safe to remove `scope` off `mutator` like this?

[the remark this answers is not in the document]
Comment (karrtikr) [reply src/vs/workbench/api/common/extHostTerminalService.ts:920]: > Is it safe to remove scope off mutator like this?

No, scope is present in the mutator type to be used in implementation.

File: src/vs/platform/terminal/common/environmentVariableCollection.ts [hunk 4]
```diff
@@ -184,9 +184,13 @@ export class MergedEnvironmentVariableCollection implements IMergedEnvironmentVa
 
 function filterScope(
 	mutator: IExtensionOwnedEnvironmentVariableMutator | IExtensionOwnedEnvironmentDescriptionMutator,
-	scope: EnvironmentVariableScope | undefined
+	scope: EnvironmentVariableScope | undefined,
+	strictFilter = false /** whether mutator has to exactly match the scope */
```
[the remark this answers is not in the document]
Comment (karrtikr) [reply src/vs/platform/terminal/common/environmentVariableCollection.ts:188]: We've to use different parameters I think, I've clarified it better in the param comment.

**Review by Tyriar — APPROVED** [review]

**Pull request merged**